server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3310,13 +3331,19 @@ public Vlan dedicatePublicIpRange(final DedicatePublicIpRangeCmd cmd) throws Res
             vlanOwner = _accountMgr.getAccount(project.getProjectAccountId());
         }
 
+        Domain domain = null;
         if (accountName != null && domainId != null) {
             vlanOwner = _accountDao.findActiveAccount(accountName, domainId);
-        }
-        if (vlanOwner == null) {
-            throw new InvalidParameterValueException("Unable to find account by name " + accountName);
-        } else if (vlanOwner.getId() == Account.ACCOUNT_ID_SYSTEM) {
-            throw new InvalidParameterValueException("Please specify a valid account. Cannot dedicate IP range to system account");
+            if (vlanOwner == null) {
```

Comment:
Looks like the project case is handled by 89ce8bd9e289327f411790eb05a9fb49ef1f9a2f. Since all tests pass, I will merge this now. Future adjustments can be made in a separate PR.